dateutil/rrule.py

```diff
@@ -447,6 +448,23 @@ def __init__(self, freq, dtstart=None,
             until = datetime.datetime.fromordinal(until.toordinal())
         self._until = until
 
+        if self._dtstart and self._until:
+            if all((
```

Comment:
I think this might be better as:

`if (self._dtstart.tzinfo is not None) != (self._until.tzinfo is not None)`

That's basically the condition under which it would fail because of the comparison anyway, and I don't want to enforce that it's specifically `tzutc` (what if you used `datetime.timezone.utc` or `pytz.UTC`). I think being slightly less strict about the specifics of the RFC is worth it in this case, since I want to tread lightly.

I'll note that there's a very strange edge case that would occur if `self._dtstart.tzinfo is self._until.tzinfo` and the UNTIL value is an ambiguous time, but I'm not terribly worried about that.

Reply:
Oh, and one thing to note is that my formulation also catches the case where *only* UNTIL is aware.

Reply:
Good point - addressed 👍